blog/2023-12-3.13-3.21.md

```diff
@@ -0,0 +1,178 @@
+<img src="/img/blog/3.13-3.21/dog-coding-laptop-mars-christmas-tree.jpg" className="border" alt="Screenshot of the new Uppy website homepage" />
+
+It’s this time of the year again! Make yourself cozy
+[by the fire](https://www.youtube.com/watch?v=ouqtX6WWL1U), and let us help you
+unwrap the gifts Uppy has prepared for you.
+
+We proud of all the improvements and amazing new features and bug fixes we were
+able to introduce this year. The highlights being: conditional S3 multipart and
+signing S3 uploads on the client, speedy handling of 10-50k file uploads,
+refresh tokens (useful when the token expires mid-upload for a large amount of
+files / slow connections), server-sent events and new `assemblyOptions` for the
+Transloadit plugin, Dashboard Single File mode (`singleFileFullScreen`),
+improved ETA in the Status Bar, a bunch of new languages, including Hindi,
+Mexican Spanish and Catalan.
+
+This one, however, is a recent Uppy updates post, covering releases from
+[`3.13.0` to `3.21.0`](https://github.com/transloadit/uppy/compare/uppy%403.13.0...uppy%403.21.0):
+announcing the switch to Typescript, Dashboard can now auto-install plugins,
+support for Catalan language, a heap of bug fixes for Tus, Companion, AWS S3,
+and more.
+
+<!--truncate-->
+
+## Typescript
+
+We are excited to announce that we are working hard to add Typescript types to
+Uppy Core and all of the plugins. We have a build setup in place that
+temporarily supports both `.ts` and `.js` files in our monorepo. And a helper
+utility that takes a package name and does the initial renaming and Typescript
+config. Then we go through the list of plugins one by one. We begun with
+`@uppy/utils` and `@uppy/core`, since most other packages depend on those. Now
+we are moving to the rest, one by one.
+
+This work so far is only affecting Uppy internals. In published packages we are
+using the manually crafted `d.ts` files, as before. We are planning to flip the
+switch and introduce some breaking changes in the next Uppy major, due to be
+released in the first quater of 2024.
+
+## Core
+
+- Allow duplicate files with `onBeforeFileAdded`
+- Fixed`onBeforeFileAdded` with Golden Retriever
+- Simplify types with class generic and type more events
+- Export `UIPlugin` and `BasePlugin` in Uppy CDN bundle
+
+## Image Editor
+
+We have a whole separate blog post about the recent rotation improvements in the
+Image Editor, so please read that, if you are interested. Other changes include:
+
+- Fixed labels on all buttons, centered the rotation slider label
+- Limited cropbox movements
+- Fixed to respect `cropperOptions.initialAspectRatio`
+- Change the checkered background color, make it semi-transparent
+- Fixed granular rotation: `[-45, 45]` instead of `[-45, 44]`
+- Removed 1px black lines
+- Made `file-editor:cancel` event fire when the Image Editor’s “cancel” button
+  is pressed
+
+<video muted autoplay loop>
+  <source src="/img/blog/3.13-3.21/crop.mov" type="video/mp4" />
+</video>
```

Comment:
nit: you technically don't need a `<source>` if you have only one
```suggestion
<video muted autoplay loop src="/img/blog/3.13-3.21/crop.mov" type="video/mp4">
</video>
```

Reply:
For sure, I just copy-pasted from previous post.